### ISSUE TITLE
inlay hints truncating end strings

### ISSUE THREAD

**Issue by gennaro-tedesco** [HampusHauffman/block.nvim#24]
Recently [inlay hints](https://github.com/neovim/neovim/pull/23984) have been merged to neovim `master` branch and are available on nightly: obviously everybody is jumping on the train!

I have noticed that, by using block.nvim, _some_ inlay hints are obfuscated, in particular by the end of the line, resulting in the end strings disappearing: see screenshot below

![Screenshot 2023-06-21 at 13 39 33](https://github.com/HampusHauffman/block.nvim/assets/15387611/70af0ad5-1f29-4174-88b4-01b2358adabd)

(you can notice that often the last part of a string, say ".nvim", is truncated). This happens with all language servers.

What can be the root cause and do you believe this is solvable in the context of this plugin?

**Comment by HampusHauffman** [comment]
Yes its probably solvable. The issue is that when you add a inlay hints the line gets longer and the block plguin that adds extmarks at the end of each line overrides this.  Not sure how to solve this in an ergonomic way atm though but it will have to be fixed eventually since as you said everyone is jumping on the train.

**Comment by IndianBoy42** [comment]
I'm not sure how you calculate the size right now but you can use `virtcol()` to get the screen column of a file position. Another plugin I had to use lsp_lines also had to be patched to account for inline virtual text: 

https://lists.sr.ht/~whynothugo/lsp_lines.nvim/patches/42031
